### PR TITLE
Better stacking

### PR DESCRIPTION
**Pull request by SubordinalBlue**
When pulling out `optima`, I found the code that used it to be rather baroque and not easy to grok.

So I thought I'd re-write them.

This is a start. Fewer LOC, and much clearer, to me at least. But I wrote this, so I'm not an impartial judge on that score.

There's more to do on this file, and I think some re-thinking of how to label/name this stuff is in order *.

Cheers.

*Some of it really assumes matrices/2D-arrays, so maybe it should be labeled and filed that way. And some could be extended or generalized to tensors of rank > 2.

_(Yeah, this PR again assumes the previous. Now that I've finally gotten to the point of writing some original code for this library, I will try to keep my PR's independent going forward.)_

### CODE REVIEW

File: src/work-stacking.lisp [hunk 1]
```diff
@@ -0,0 +1,58 @@
+;;;; Scratch work for 'better' stacking functions.
```
Comment (bendudson) [inline src/work-stacking.lisp:1]: Can this file be deleted from the repo?

File: src/stacking.lisp [hunk 2]
```diff
@@ -18,59 +40,56 @@ This function should be used to implement copying of contiguous row-major blocks
...
-          (mapc (lambda (start-rows-and-dims object)
-                  (destructuring-bind (start-row &rest dims)
-                      start-rows-and-dims
-                    (if dims
-                        (stack-rows-copy object result element-type start-row)
-                        (fill (displace result ncol (* start-row ncol))
-                              (coerce object element-type)))))
-                start-rows-and-dims objects)
-          result)))))
+  "Stack OBJECTS, row-wise, into a simple 2D array of given ELEMENT-TYPE.
```
Comment (bendudson) [inline src/stacking.lisp:62]: Thanks! This is much clearer to me as well. I think I actually understand what your code is doing...

File: src/work-stacking.lisp (hunk 1 above)
Comment (SubordinalBlue) [reply src/work-stacking.lisp:1]: Oops.

Oh yeah, that'll be gone before I convert this from a Draft PR to a normal PR.

File: src/stacking.lisp (hunk 2 above)
Comment (SubordinalBlue) [reply src/stacking.lisp:62]: Glad to hear it. :-)

Like nearly every thing re-written, I started out sure I could make it very clean and concise.
Then, I realized all the corner cases, nuances, etc. :-)

Comment (bendudson) [reply src/stacking.lisp:62]: As long as we have the unit tests to catch the corner cases etc. then all should be well. Thanks for this!

**Comment by Symbolics** [comment]
Wow. 75 commits in a single PR. That's got to be some kind of record. Is this still a work in progress?

**Comment by bendudson** [comment]
Sorry for the long delay. Closing this as the new home for array-operations is https://github.com/Lisp-Stat/array-operations
